app/models/component.rb

```diff
@@ -75,4 +74,24 @@ def type
   def display
     I18n.t("user_journey.component_name.#{type}")
   end
+
+private
+
+  def encryption_certificate_check
+    x509 = encryption_certificate&.x509
+    if x509.nil? || x509.not_after < Time.now
+      Rails.logger.error "When publishing the meta data the service '#{name}' has been identified as having an expired encryption certificate."
+      nil
+    else
+      encryption_certificate
+    end
+  end
+
+  def enabled_signing_certificates_check
+    non_expiring_certs = enabled_signing_certificates.reject { |cert| cert.x509.not_after < Time.now }
```

Comment:
~non_expiring_certs~ => `valid_certs`  ?

Reply:
Strictly speaking, we're only checking that they've not expired, not that they're valid in every way (e.g. OCSP).